test/JoinTest.cpp

```diff
@@ -230,11 +238,27 @@ using ExpectedColumns = ad_utility::HashMap<
     std::pair<std::span<const Id>, ColumnIndexAndTypeInfo::UndefStatus>>;
 
 // Test that the result of the `join` matches the `expected` outcome.
-void testJoinOperation(Join& join, const ExpectedColumns& expected) {
-  auto res = join.getResult();
+void testJoinOperation(Join& join, const ExpectedColumns& expected,
+                       bool requestLaziness = false,
+                       bool expectLazinessParityWhenNonEmpty = false,
```

Comment:
The name `...ParityWhenNonEmpty` is not self-explanatory, please document and/or rename it.